_posts/2019-11-25-query-pulsar-streams-using-apache-flink.md

```diff
@@ -0,0 +1,220 @@
+
+
+# Leveraging the Flink <> Pulsar Schema Integration
+
+Before delving into the integration details and how you can use Pulsar schema with Flink, let us describe how schema in Pulsar works. Schema in Apache Pulsar already co-exists and serves as the representation of the data on the broker side of the framework, something that makes schema registry with external systems obsolete. Additionally, the data schema in Pulsar is associated with each topic so both producers and consumers send data with predefined schema information, while the broker performs schema validation, and manages schema multi-versioning and evolution in compatibility checks. 
+ 
+Below you can find an example of Pulsar’s schema on both the producer and consumer side. On the producer side, you can specify which schema you want to use and Pulsar then sends a POJO class without the need to perform any serialization/deserialization. Similarly, on the consumer end, you can also specify the data schema and upon receiving the data, Pulsar will automatically validate the schema information, fetch the schema of the given version and then deserialize the data back to a POJO structure. Pulsar stores the schema information in the metadata of a Pulsar topic.
+
+```java
+// Create producer with Struct schema and send messages
+Producer<User> producer = client.newProducer(Schema.AVRO(User.class)).create();
+producer.newMessage()
+  .value(User.builder()
+    .userName(“pulsar-user”)
+    .userId(1L)
+    .build())
+  .send();
+```
+
+```java
+// Create consumer with Struct schema and receive messages
+Consumer<User> consumer = client.newCOnsumer(Schema.AVRO(User.class)).create();
+consumer.receive();
+```
+
+Let’s assume we have an application that specifies a schema to the producer and/or consumer. Upon receiving the schema information, the producer (or consumer) — that is connected to the broker — will transfer such information so that the broker can then perform schema registration, validations and schema compatibility checks before returning or rejecting the schema as illustrated in the diagram below: 
+
+<center>
+<img src="{{ site.baseurl }}/img/blog/flink-pulsar-sql-blog-post-visual.png" width="600px" alt="Pulsar Schema"/>
```

Comment:
I can not merge this because the file is missing.

Reply:
thanks @rmetzger - I added both files.